common/src/main/java/io/netty/util/concurrent/PromiseNotifier.java

```diff
@@ -61,6 +61,57 @@ public PromiseNotifier(boolean logNotifyFailure, Promise<? super V>... promises)
         this.logNotifyFailure = logNotifyFailure;
     }
 
+    /**
+     * Link the {@link Future} and {@link Promise}. This means that if the {@link Future} completes the {@link Promise}
+     * will be notified. That said cancellation is propagated both ways. This means if the {@link Future} is cancelled
+     * the {@link Promise} is cancelled as well and vise-versa.
+     *
+     * @param future    the {@link Future} which will be used to listen to for notifying the {@link Promise}.
+     * @param promise   the {@link Promise} which will be notified
+     * @param <V>       the type of the value.
+     * @param <F>       the type of the {@link Future}
+     * @return          the passed in {@link Future}
+     */
+    public static <V, F extends Future<V>> F link(final F future, final Promise<? super V> promise) {
+        return link(true, future, promise);
+    }
+
+    /**
+     * Link the {@link Future} and {@link Promise}. This means that if the {@link Future} completes the {@link Promise}
+     * will be notified. That said cancellation is propagated both ways. This means if the {@link Future} is cancelled
+     * the {@link Promise} is cancelled as well and vise-versa.
+     *
+     * @param logNotifyFailure  {@code true} if logging should be done in case notification fails.
+     * @param future            the {@link Future} which will be used to listen to for notifying the {@link Promise}.
+     * @param promise           the {@link Promise} which will be notified
+     * @param <V>               the type of the value.
+     * @param <F>               the type of the {@link Future}
+     * @return                  the passed in {@link Future}
+     */
+    @SuppressWarnings({"unchecked", "rawtypes"})
+    public static <V, F extends Future<V>> F link(boolean logNotifyFailure, final F future,
```

Comment:
nit: I think `cascade` is a better name as it is "cascading" a value from one future to another.